### ISSUE TITLE
CFQUERYPARAM_REQ false positive: String in SQL Server XML function

### ISSUE THREAD

**Issue by TheRealAgentK** [cflint/CFLint#297]
```
...
UPDATE
					Something
				SET
					taskXML.modify('insert #additionalMember#
									   as last into (/task)[1]')
				WHERE
					tID = <cfqueryparam value="#arguments.something.tID#" cfsqltype="cf_sql_integer">
					AND taskUUID = <cfqueryparam value="#arguments.something.tUUID#" cfsqltype="cf_sql_varchar">
...
```

Reports on the taskXML.modify (taskXML is a column from a preceeding IF EXISTS ( SELECT... ))

**Comment by ryaneberly** [comment]
@TheRealAgentK ,  can you expand on this one a bit please.   I get a single CFQUERYPARAM_REQ because of #additionalMember# as I would expect.

**Comment by TheRealAgentK** [comment]
Will try to create and then push an isolated test case in dev.

**Comment by TheRealAgentK** [comment]
Ah, scratch that - I see what you mean now. Yes, that is a valid one because .modify() actually acts like an update on that XML blurb. It's another one that's not fixable with CFQUERYPARAM though and would benefit from the more fine-grained ignoring in SQL.

**Comment by ryaneberly** [comment]
Thank you.

**Comment by zspitzer** [comment]
this is also triggered when the case of the <cfquery> and <cfqueryparam> tags don't match, which is only stylistically messy but still valid

**Comment by TheRealAgentK** [comment]
Hey @zspitzer - can you provide a (couple of) test samples for us?

**Comment by zspitzer** [comment]
here's a simple test case (attached too)

`<cfset user_id=1>`
`<CFquery name="users" datasource="dsn">`
	SELECT 	user
	FROM 	users
	where 	user_id = `<CFqueryparam value="#user_id#" cfsqltype="CF_SQL_NUMERIC">`
`</CFquery>`

 java -jar .\CFLint-1.2.0-all.jar -folder . -stdout -text
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
Issue
Severity:WARNING
Message code:CFQUERYPARAM_REQ
        File: queryparam_req.cfm
        Column:0
        Line:5
                Message:`<cfquery>` should use `<cfqueryparam/>` for variable 'user_id'.
                Variable:'user_id' in function:
                Expression:`<CFquery name="users" datasource="dsn">`
        SELECT  user
        FROM    users
        where   user_id = `<CFqueryparam value="#user_id#" cfsqltype="CF_SQL_NUMERIC">`
`</CFquery>`


Total files:1
Total lines:7

Issue counts:1
CFQUERYPARAM_REQ:1

Total issues:1
Total warnings:1


[queryparam_req.cfm.txt](https://github.com/cflint/CFLint/files/1184981/queryparam_req.cfm.txt)

**Comment by zspitzer** [comment]
done #380
